gapis/api/vulkan/framegraph.go

```diff
@@ -82,25 +107,61 @@ func newFramegraphImage(state *State, img *ImageObjectʳ) *api.FramegraphImage {
 	memoryMapped := false
 	for _, planeMemInfo := range img.PlaneMemoryInfo().All() {
 		mem := planeMemInfo.BoundMemory()
-		coherentMemory = coherentMemory || backedByCoherentMemory(state, mem)
-		memoryMapped = memoryMapped || !(mem.MappedLocation().IsNullptr())
+		if mem != NilDeviceMemoryObjectʳ {
+			coherentMemory = coherentMemory || backedByCoherentMemory(state, mem)
+			if !(mem.MappedLocation().IsNullptr()) {
+				memRange := memory.Range{
+					Base: uint64(planeMemInfo.BoundMemoryOffset()),
+					Size: uint64(mem.AllocationSize()),
+				}
+				mappedRange := memory.Range{
+					Base: uint64(mem.MappedOffset()),
+					Size: uint64(mem.MappedSize()),
+				}
+				memoryMapped = memoryMapped || memRange.Overlaps(mappedRange)
+			}
+		}
 	}
 	// No need to scan sparse memory if both coherent/mapped are already true
 	if !coherentMemory || !memoryMapped {
 		for _, sparseMemBinding := range img.OpaqueSparseMemoryBindings().All() {
 			mem := state.DeviceMemories().Get(sparseMemBinding.Memory())
-			coherentMemory = coherentMemory || backedByCoherentMemory(state, mem)
-			memoryMapped = memoryMapped || !(mem.MappedLocation().IsNullptr())
+			if mem != NilDeviceMemoryObjectʳ {
+				coherentMemory = coherentMemory || backedByCoherentMemory(state, mem)
+				if !(mem.MappedLocation().IsNullptr()) {
+					memRange := memory.Range{
+						Base: uint64(sparseMemBinding.MemoryOffset()),
+						Size: uint64(sparseMemBinding.Size()),
+					}
+					mappedRange := memory.Range{
+						Base: uint64(mem.MappedOffset()),
+						Size: uint64(mem.MappedSize()),
+					}
+					memoryMapped = memoryMapped || memRange.Overlaps(mappedRange)
+				}
+			}
 		}
 	}
 	if !coherentMemory || !memoryMapped {
 		for _, sparseImgMem := range img.SparseImageMemoryBindings().All() {
-			for _, layers := range sparseImgMem.Layers().All() {
-				for _, level := range layers.Levels().All() {
-					for _, blocks := range level.Blocks().All() {
-						mem := state.DeviceMemories().Get(blocks.Memory())
-						coherentMemory = coherentMemory || backedByCoherentMemory(state, mem)
-						memoryMapped = memoryMapped || !(mem.MappedLocation().IsNullptr())
+			for _, layer := range sparseImgMem.Layers().All() {
+				for _, level := range layer.Levels().All() {
+					for _, block := range level.Blocks().All() {
+						mem := state.DeviceMemories().Get(block.Memory())
+						if mem != NilDeviceMemoryObjectʳ {
+							coherentMemory = coherentMemory || backedByCoherentMemory(state, mem)
+							if !(mem.MappedLocation().IsNullptr()) {
+								memRange := memory.Range{
+									Base: uint64(block.MemoryOffset()),
+									Size: uint64(block.Size()),
+								}
+								mappedRange := memory.Range{
+									Base: uint64(mem.MappedOffset()),
+									Size: uint64(mem.MappedSize()),
+								}
+								memoryMapped = memoryMapped || memRange.Overlaps(mappedRange)
```

Comment:
You should be able to drop out of this nested loop if coherent&&mapped.

Reply:
Added early exits in loops.